hedera-node/src/main/java/com/hedera/services/throttling/DeterministicThrottling.java

```diff
@@ -130,10 +157,32 @@ public void rebuildFor(ThrottleDefinitions defs) {
 
 		functionReqs = newFunctionReqs;
 		activeThrottles = newActiveThrottles;
-
+		applyGasConfig();
 		logResolvedDefinitions();
 	}
 
+	@Override
+	public void applyGasConfig() {
+		if (consensusThrottled) {
+			if (dynamicProperties.getConsensusThrottleMaxGasLimit() == 0 && dynamicProperties.shouldThrottleByGas()) {
+				log.error("ThrottleByGas global dynamic property is set to true but contracts.consensusThrottleMaxGasLimit is not set in throttles.json or is set to 0.");
+			} else {
+				gasThrottle = new GasLimitDeterministicThrottle(dynamicProperties.getConsensusThrottleMaxGasLimit());
```

Comment:
Shouldn't this constructor argument be divide by `n` from L140, `n = capacitySplitSource.getAsInt()`?

Otherwise we will not get the desired effect of "splitting" the capacity of the HAPI throttle evenly across all network nodes.